Sync/Synchronizers/ClientsSynchronizer.swift

```diff
@@ -320,6 +322,8 @@ open class ClientsSynchronizer: TimestampedSingleCollectionSynchronizer, Synchro
             }
 
             if rec.id == ourGUID {
+                log.debug("Skipping our own client records since the guid is not the fxa device ID")
```

Comment:
This should probably be a warning, or maybe even sentry, as it implies we failed to migrate. Actually, it does look like you still do get the "old" record on the first sync - it would probably be ideal if `self.maybeDeleteClients` somehow managed to remove the old guid it found, but that might not be trivial - but in that scenario, I guess you new `hasLocalClientIdChanged` will be true (or the timestamp will be zero if you take the idea above), so you can probably still make noise if you have the guid, but don't have the flag set that tells you it was migrated.